src/wallet/wallet.cpp

```diff
@@ -4286,24 +4306,19 @@ util::Result<MigrationResult> MigrateLegacyToDescriptor(const std::string& walle
     std::set<fs::path> wallet_dirs;
     if (success) {
         // Migration successful, unload all wallets locally, then reload them.
-        const auto& reload_wallet = [&](std::shared_ptr<CWallet>& to_reload) {
-            assert(to_reload.use_count() == 1);
-            std::string name = to_reload->GetName();
-            wallet_dirs.insert(fs::PathFromString(to_reload->GetDatabase().Filename()).parent_path());
-            to_reload.reset();
-            to_reload = LoadWallet(context, name, /*load_on_start=*/std::nullopt, options, status, error, warnings);
-            return to_reload != nullptr;
-        };
         // Reload the main wallet
+        wallet_dirs.insert(fs::PathFromString(local_wallet->GetDatabase().Filename()).parent_path());
```

Comment:
In commit "wallet: Reload the wallet if migration exited early" (78ba0e6748d2a519a96c41dea851e7c43b82f251)

Not related to this PR, but I think the `fs::PathFromString(wallet->GetDatabase().Filename()).parent_path()` expression repeated throughout this function is unnecessarily verbose and also potentially dangerous if it is used in another context, or if the surrounding context changes.

 It happens to be safe currently, because we know this code is only called after MigrateToSQLite is called, so the database `FileName()` method will always return the path to a file in a wallet subdirectory. But in other contexts `wallet->GetDatabase().Filename()` could return the path to a file in a top level wallets directory, or even a top level data directory, so calling `fs::remove_all(fs::PathFromString(wallet->GetDatabase().Filename()).parent_path())` could wipe out a lot of other data not part of the wallet we are trying to delete.

To fix this as a start, I think we should be deleting the original wallet path
`AbsPathJoin(GetWalletDir(), fs::PathFromString(wallet->GetName()))`, not the parent of the path of the database inside the wallet. It would also be good to move away from using `fs::remove_all` which could delete a lot of files unintentially, and instead write a wallet delete function that only deletes files we know about.